example/10_conv3d_fwd_xdl/conv3d_fwd_xdl.cpp

```diff
@@ -0,0 +1,281 @@
+        8,              // ABlockTransferDstScalarPerVector_K1
+        true,           // ABlockLdsAddExtraM
+        S<4, 64, 1>,    // BBlockTransferThreadClusterLengths_K0_N_K1
+        S<1, 0, 2>,     // BBlockTransferThreadClusterArrangeOrder
+        S<1, 0, 2>,     // BBlockTransferSrcAccessOrder
+        2,              // BBlockTransferSrcVectorDim
+        8,              // BBlockTransferSrcScalarPerVector
+        8,              // BBlockTransferDstScalarPerVector_K1
+        true,           // BBlockLdsAddExtraN
+        7,              // CThreadTransferSrcDstVectorDim
+        1>;             // CThreadTransferDstScalarPerVector
+
+int main(int argc, char* argv[])
+{
+    bool do_verification = false;
+    int init_method      = 0;
+    int nrepeat          = 5;
+
+    // convolution shape
+    ck::index_t N                                   = 4;
+    ck::index_t K                                   = 256;
+    ck::index_t C                                   = 192;
+    std::vector<ck::index_t> in_spatial_lengths     = {71, 71, 71};
+    std::vector<ck::index_t> filter_spatial_lengths = {3, 3, 3};
+    std::vector<ck::index_t> conv_filter_strides    = {2, 2, 2};
+    std::vector<ck::index_t> conv_filter_dilations  = {1, 1, 1};
+    std::vector<ck::index_t> in_left_pads           = {1, 1, 1};
+    std::vector<ck::index_t> in_right_pads          = {1, 1, 1};
+
+    if(argc == 4)
+    {
+        do_verification = std::stoi(argv[1]);
+        init_method     = std::stoi(argv[2]);
+        nrepeat         = std::stoi(argv[3]);
+    }
+    else if(argc == 25)
+    {
+        do_verification = std::stoi(argv[1]);
+        init_method     = std::stoi(argv[2]);
+        nrepeat         = std::stoi(argv[3]);
+
+        N                         = std::stoi(argv[4]);
+        K                         = std::stoi(argv[5]);
+        C                         = std::stoi(argv[6]);
+        filter_spatial_lengths[0] = std::stoi(argv[7]);
+        filter_spatial_lengths[1] = std::stoi(argv[8]);
+        filter_spatial_lengths[2] = std::stoi(argv[9]);
+        in_spatial_lengths[0]     = std::stoi(argv[10]);
+        in_spatial_lengths[1]     = std::stoi(argv[11]);
+        in_spatial_lengths[2]     = std::stoi(argv[12]);
+        conv_filter_strides[0]    = std::stoi(argv[13]);
+        conv_filter_strides[1]    = std::stoi(argv[14]);
+        conv_filter_strides[2]    = std::stoi(argv[15]);
+        conv_filter_dilations[0]  = std::stoi(argv[16]);
+        conv_filter_dilations[1]  = std::stoi(argv[17]);
+        conv_filter_dilations[2]  = std::stoi(argv[18]);
+        in_left_pads[0]           = std::stoi(argv[19]);
+        in_left_pads[1]           = std::stoi(argv[20]);
+        in_left_pads[2]           = std::stoi(argv[21]);
+        in_right_pads[0]          = std::stoi(argv[22]);
+        in_right_pads[1]          = std::stoi(argv[23]);
+        in_right_pads[2]          = std::stoi(argv[24]);
```

Comment:
You can use getopt_long() to help classify and analyze the 25 arguments

Reply:
Good to know. But since other examples don't use it, I'll leave it as it. We can change them all together in future if needed.